yamux/src/connection.rs

```diff
@@ -359,6 +362,13 @@ struct Active<T> {
 pub(crate) enum StreamCommand {
     /// A new frame should be sent to the remote.
     SendFrame(Frame<Either<Data, WindowUpdate>>),
+    Flush {
+        id: StreamId,
+        /// How many frames we've queued for sending at the time the flush was requested.
+        num_frames: u64,
```

Comment:
Wondering: Do we need to send this? We could also just send the waker and simply compare the counters in `Shared`. That will be done anyway by the  implementation of `poll_flush` so we might as well use it in the connection task.

I think that makes this a bit easier to understand.

Reply:
Doubling down on this: I think this means we can contain the state entirely within `Shared` and only store a single `Waker` for flushing.

Reply:
I tried for a bit but can't seem to make a solution that doesn't include this flush mark.